rpy2/tests/rinterface/test_embedded_r.py

```diff
@@ -179,24 +181,26 @@ def testInterruptR():
         import sys
         sys.path.insert(0, '%s')
         import rpy2.rinterface as ri
+        from rpy2.rinterface_lib import callbacks
+        from rpy2.rinterface_lib import embedded
 
         ri.initr()
         def f(x):
             pass
-        ri.set_writeconsole_regular(f)
```

Comment:
I vaguely remember that this was added to keep the output from being too verbose. Ideally there should be a mechanism to restore the the original print callback to its original setting no matter the outcome of evaluating the code though.

Let me run the automated test to see if I am correct on any of this.

Reply:
Ah, that completely makes sense. Do you think the below replacement of `consolewrite_print` with `f` will accomplish this? From a little grepping, I had the hunch `set_writeconsole_regular` was deprecated and perhaps replaced by `consolewrite_print`, but was not sure. Also seems like the `obj_in_module` [resets the parameter outside the context manager](https://github.com/rpy2/rpy2/blob/384dddcba29aab774d4657c1c895150524a0c30f/rpy2/rinterface_lib/callbacks.py#L20-L27), so I hope that would address your second concern?

Reply:
I think the test might not have been working in the past which I tried to trace out in the commit message for https://github.com/rpy2/rpy2/pull/832/commits/18ae4ff3dab08c17368d76ba768465b8512d5804. Seems like there were two problems; the dedent issue as well as this `set_writeconsole_regular` issue. It was resulting in a non-zero exit code, but because the test wasn't checking for a specific exit code (the assertion was `assert ret_code is not None`) it wasn't able to distinguish process interruption from these errors.